titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/admission/ServiceMeshImageSanitizer.java

```diff
@@ -0,0 +1,212 @@
+import com.netflix.titus.api.jobmanager.model.job.Image;
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor;
+import com.netflix.titus.api.jobmanager.model.job.JobFunctions;
+import com.netflix.titus.common.model.admission.AdmissionSanitizer;
+import com.netflix.titus.common.model.admission.ValidatorMetrics;
+import com.netflix.titus.common.util.StringExt;
+import com.netflix.titus.runtime.connector.registry.RegistryClient;
+import com.netflix.titus.runtime.connector.registry.TitusRegistryException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.time.Duration;
+import java.util.function.UnaryOperator;
+
+/**
+ * This {@link AdmissionSanitizer} implementation validates and sanitizes service mesh image attributes.
+ */
+@Singleton
+public class ServiceMeshImageSanitizer implements AdmissionSanitizer<JobDescriptor> {
+    private static final Logger logger = LoggerFactory.getLogger(ServiceMeshImageSanitizer.class);
+
+    private final ServiceMeshImageSanitizerConfiguration configuration;
+    private final RegistryClient registryClient;
+    private final ValidatorMetrics validatorMetrics;
+
+    @Inject
+    public ServiceMeshImageSanitizer(ServiceMeshImageSanitizerConfiguration configuration, RegistryClient registryClient, Registry spectatorRegistry) {
+        this.configuration = configuration;
+        this.registryClient = registryClient;
+        this.validatorMetrics = new ValidatorMetrics(this.getClass().getSimpleName(), spectatorRegistry);
+    }
+
+    /**
+     * @return a {@link UnaryOperator} that adds a sanitized Image or job attributes when sanitization was skipped
+     */
+    @Override
+    public Mono<UnaryOperator<JobDescriptor>> sanitize(JobDescriptor jobDescriptor) {
+        if (isDisabled()) {
+            return Mono.just(ServiceMeshImageSanitizer::skipSanitization);
+        }
+
+        if (!serviceMeshIsEnabled(jobDescriptor)) {
+            validatorMetrics.incrementValidationSkipped("serviceMeshNotEnabled");
+            return Mono.just(UnaryOperator.identity());
+        }
+
+        if (!serviceMeshIsPinned(jobDescriptor)) {
+            validatorMetrics.incrementValidationSkipped("serviceMeshNotPinned");
+            return Mono.just(UnaryOperator.identity());
+        }
+
+        try {
+            Image image = getServiceMeshImage(jobDescriptor);
+            return sanitizeServiceMeshImage(image)
+                    .map(ServiceMeshImageSanitizer::setMeshImageFunction)
+                    .timeout(Duration.ofMillis(configuration.getServiceMeshImageValidationTimeoutMs()))
+                    .doOnSuccess(j -> validatorMetrics.incrementValidationSuccess(image.getName()))
+                    .onErrorReturn(throwable -> isAllowedException(throwable, image), ServiceMeshImageSanitizer::skipSanitization);
+        } catch (Throwable t) {
+            return Mono.error(t);
```

Comment:
@tbak I want to call this change to your attention, in case you missed it

Reply:
Yes, I have seen it. It is a good safety net.